biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -1713,7 +1713,9 @@ void augmentImports(Packages imports, Packages exports) throws Exception {
 					exportVersion = cleanupVersion(exportVersion);
 
 					importRange = applyVersionPolicy(exportVersion, importRange, provider);
-					importAttributes.put(VERSION_ATTRIBUTE, importRange);
+					if(importRange == null || !importRange.trim().isEmpty()) {
```

Comment:
I don't see how `applyVersionPolicy` could ever return null. It is possible for it to return an empty string if the input importRange is null and `-provider-policy` or `-consumer-policy` is set to the empty string.

So a null check is not necessary. Just a check that importRange is empty.